src/server/pps/server/api_server.go

```diff
@@ -926,66 +866,50 @@ func (a *apiServer) StopJobInTransaction(txnCtx *txncontext.TransactionContext,
 	if reason == "" {
 		reason = "job stopped"
 	}
-	return a.stopJob(txnCtx, request.Job, request.OutputCommit, reason)
+	return a.stopJob(txnCtx, request.Job, reason)
 }
 
-func (a *apiServer) stopJob(txnCtx *txncontext.TransactionContext, job *pps.Job, outputCommit *pfs.Commit, reason string) error {
+func (a *apiServer) stopJob(txnCtx *txncontext.TransactionContext, job *pps.Job, reason string) error {
 	jobs := a.jobs.ReadWrite(txnCtx.SqlTx)
-	if (job == nil) == (outputCommit == nil) {
-		return errors.New("Exactly one of Job or OutputCommit must be specified")
+	if job == nil {
+		return errors.New("Job must be specified")
 	}
 
 	jobInfo := &pps.StoredJobInfo{}
-	if job != nil {
-		if err := jobs.Get(job.ID, jobInfo); err != nil {
-			return err
-		}
-		outputCommit = jobInfo.OutputCommit
+	if err := jobs.Get(ppsdb.JobKey(job), jobInfo); err != nil {
+		return err
 	}
 
 	commitInfo, err := a.env.PfsServer().InspectCommitInTransaction(txnCtx, &pfs.InspectCommitRequest{
-		Commit: proto.Clone(outputCommit).(*pfs.Commit),
+		Commit: jobInfo.OutputCommit,
 	})
 	if err != nil && !pfsServer.IsCommitNotFoundErr(err) && !pfsServer.IsCommitDeletedErr(err) {
 		return err
 	}
+
 	if commitInfo != nil {
 		if err := a.env.PfsServer().FinishCommitInTransaction(txnCtx, &pfs.FinishCommitRequest{
 			Commit: commitInfo.Commit,
 			Empty:  true,
 		}); err != nil && !pfsServer.IsCommitNotFoundErr(err) && !pfsServer.IsCommitDeletedErr(err) && !pfsServer.IsCommitFinishedErr(err) {
 			return err
 		}
-		statsCommit, err := ppsutil.GetStatsCommit(commitInfo)
```

Comment:
Just a note, be sure to account for the case where the job is for a service pipeline which doesn't have a meta commit.

Reply:
This PR has the change / more info: https://github.com/pachyderm/pachyderm/pull/6232

Reply:
So this function definitely gracefully handles commits not being found (it biases towards just getting everything to a closed state rather than erroring if something isn't found).